### DISCUSSION TITLE
Versioning switching on same page behaviour

### DISCUSSION THREAD

**Discussion by lennizle** [squidfunk/mkdocs-material#4521] [Q&A]
I basically have the same problem that was reported here: https://github.com/squidfunk/mkdocs-material/discussions/3617

I would like to have version switching on the same page (if available) through outdated banner OR version switch

In both cases clicking outdated banner and version switch, i am redirected to the root page of the respective version. Is that the intended behavior? From what I see in the example page (https://squidfunk.github.io/mkdocs-material-example-versioning/0.1/bar/) same page version switch should work.

Reproducable with minimal setup:

```
mike deploy 1.0
mike deploy 1.1 latest
mike set-default latest
mike serve
```

Package versions:
```
Python: 3.10.0
MkDocs: 1.4.1
Mike: 1.1.2
Material: 8.5.6
```

**Comment by squidfunk** [comment]
Thanks for reporting. Please provide a reproducible self-contained example that I can run (or deploy), so I can look into it. Also, please provide the exact steps you're executing and exactly what you're trying to achieve.

**Comment by lennizle** [comment]
Hey thanks for answering and your great project.

I created an example and the exact steps to reproduce. Please have a look here: https://github.com/lennizle/mkdocs-material-example

Cheers to Cologne from Düsseldorf :)

**Comment by lennizle** [comment]
@squidfunk were you able to reproduce?

**Reply by squidfunk** [reply]
Haven't had the time to yet, sorry.

**Reply by squidfunk** [reply]
Have you tried cloning and deploying [our versioning example](https://github.com/squidfunk/mkdocs-material-example-versioning) with the steps mentioned in the README? Does the error happen then as well, or is it only with your own reproduction? Additionally, does it only happen with `mike serve` or when deploying and checking the production site?

**Reply by squidfunk** [reply]
The reason why I'm asking – if you try a project actually using versioning, e.g. [hatch](https://hatch.pypa.io/1.6/config/build/), you see that it works as intended.
